build-tools/packages/build-cli/src/commands/generate/sourceEntrypoints.ts

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { PackageJson } from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import type { TsConfigJson } from "type-fest";
+import {
+	generateSourceEntrypoints,
+	optionDefaults,
+	readPackageJson,
+	readTsConfig,
+} from "../../library/commands/generateSourceEntrypoints.js";
+import { ApiTag, BaseCommand } from "../../library/index.js";
+// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+// eslint-disable-next-line import/no-internal-modules
+import { type ExportData, mapExportPathsFromPackage } from "../../library/packageExports.js";
+import type { CommandLogger } from "../../logging.js";
+
+/**
+ * Generates source entrypoints for Fluid Framework APIs to support API levels (/alpha, /beta. etc.).
+ */
+export default class GenerateSourceEntrypointsCommand extends BaseCommand<
+	typeof GenerateSourceEntrypointsCommand
+> {
+	static readonly description =
+		`Generates TypeScript source files that roll up APIs into different entrypoint files. The entrypoints are determined based on the "exports" field in package.json."`;
+
+	static readonly flags = {
+		mainEntrypoint: Flags.file({
+			description: "Main entrypoint file containing all untrimmed exports.",
+			default: optionDefaults.mainEntrypoint,
+			exists: true,
+		}),
+		...BaseCommand.flags,
+	};
+
+	public async run(): Promise<void> {
+		const { mainEntrypoint } = this.flags;
+
+		const packageJson = await readPackageJson();
+
+		const tsConfig = await readTsConfig();
+
+		const { outFileSuffix, outFileAlpha, outFileBeta, outFileLegacy, outFilePublic, outDir } =
+			optionDefaults;
+
+		const mapSrcQueryPathToApiTagLevel: Map<string | RegExp, ApiTag | undefined> = new Map([
+			[`${outDir}${outFileAlpha}${outFileSuffix}`, ApiTag.alpha],
+			[`${outDir}${outFileBeta}${outFileSuffix}`, ApiTag.beta],
+			[`${outDir}${outFilePublic}${outFileSuffix}`, ApiTag.public],
+			[`${outDir}${outFileLegacy}${outFileSuffix}`, ApiTag.legacy],
+		]);
+
+		const mapSrcApiTagLevelToOutput = getOutputConfiguration(
+			packageJson,
+			tsConfig,
+			mapSrcQueryPathToApiTagLevel,
+			this.logger,
+		);
+
+		if (mapSrcApiTagLevelToOutput.size === 0) {
+			throw new Error(
+				`There are no package exports matching requested output entrypoints:\n\t${[
+					...mapSrcQueryPathToApiTagLevel.keys(),
+				].join("\n\t")}`,
+			);
+		}
+
+		const promises: Promise<void>[] = [];
+		promises.push(
+			generateSourceEntrypoints(mainEntrypoint, mapSrcApiTagLevelToOutput, this.logger),
+		);
+
+		// All of the output actions (deletes of stale files or writing of new/updated files)
+		// are all independent and can be done in parallel.
+		await Promise.all(promises);
```

Comment:
The array only has a single item, so no need for Promise.all. Did you mean for this to be in a loop of some kind?